src/component/BookList.js

```diff
@@ -1,14 +1,19 @@
-import { useSelector } from 'react-redux';
+import './styles/book.css';
+import { useSelector, useDispatch } from 'react-redux';
+import React, { useEffect } from 'react';
 import Book from './Book';
+import { fetchBooks } from '../redux/book/booksSlice';
 
 export default function BookList() {
   const { totalbooks } = useSelector((state) => state.book);
-  const updateid = totalbooks.map((b) => ({
-    ...b,
-  }));
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
   return (
-    <ul>
-      {updateid.map((item) => (
+    <ul className="booklist">
```

Comment:
👍 Good job with rendering all the books! However, there are a couple of areas where you can improve your project's visual alignment with the design:

- [ ] Gap Between Books: The gap between two book cards currently doesn't match the design. To achieve a design-compliant appearance, please consider adjusting the gap between the book cards to match the desired layout.

- [ ] Card List Background: The background of the card list doesn't match the design. Please consider updating your code to match the design's background styling.

| Your | Design |
| --- | --- |
|  ![image](https://github.com/MayPyone/Bookstore/assets/55840999/b1f6eb97-351e-4084-b018-d17781ae7faf)| ![image](https://github.com/MayPyone/Bookstore/assets/55840999/25a79e18-8a95-4aa5-a295-aa72508addc8)|


By making these adjustments, you'll ensure that your project aligns more closely with the design specifications. Keep up the great work! 🌟📚